src/docs/framework-integration/vue.md

```diff
@@ -8,50 +8,378 @@ contributors:
   - kensodemann
   - brysalazar12
   - iskanderbroere
+  - sean-perkins
 ---
 
 # Vue
 
-In order to use the custom element library within the Vue app, the application must be modified to define the custom elements and to inform the Vue compiler which elements to ignore during compilation. This can all be done within the `main.js` file.
+**Supports: Vue 3 • TypeScript 4.0+ • Stencil v2.9.0+**
 
-Assuming you’ve run `npm install --save test-components` beforehand, and that `test-component` is the name of our made up Web Components that we have published to npm, you import the components into the 'main.js' file by
+Stencil can generate Vue component wrappers for your web components. This allows your Stencil components to be used within a Vue 3 application. The benefits of using Stencil's component wrappers over the standard web components include:
 
-- importing the node module
-- telling Vue to ignore the custom element tags (see [docs](https://vuejs.org/v2/api/#ignoredElements))
-- binding the Stenciljs component code to the window object
+- Type checking with your components.
+- Integration with the router link and Vue router.
+- Optionally, form control web components can be used with `v-model`.
 
-```tsx
-import Vue from 'vue';
-import App from './App.vue';
+## Setup
 
-import { applyPolyfills, defineCustomElements } from 'test-components/loader';
+### Project Structure
 
-Vue.config.productionTip = false;
+We recommend using a monorepo structure for your component library with component wrappers. Your project workspace should contain your Stencil component library and the library for the generate Vue component wrappers.
 
-// Tell Vue to ignore all components defined in the test-components
-// package. The regex assumes all components names are prefixed
-// 'test'
-Vue.config.ignoredElements = [/test-\w*/];
+An example project set-up may look similar to:
 
-// Bind the custom elements to the window object
-applyPolyfills().then(() => {
-  defineCustomElements();
-});
+```
+top-most-directory/
+└── packages/
+    ├── vue-library/
+    │   └── src/
+    │       ├── lib/
+    │       └── index.ts
+    └── stencil-library/
+        ├── stencil.config.js
+        └── src/components
+```
+
+#### Creating a Stencil Component Library
+
+> If you already have a Stencil component library, skip this section.
+
+```bash
+cd packages/
+npm init stencil components stencil-library
+cd stencil-library
+# Install dependencies
+npm install
+# of if you are using yarn
+yarn install
+cd ..
+lerna bootstrap
```

Comment:
We probably shouldn't assume any specific monorepo tooling. Folks using nx, bazel, etc. can probably figure out their equivalent command, but if someone's not coming from a monorepo tooling background and this is their first experience with it, they may be stuck scratching their heads here. My recommendation is we do a tooling-agnostic package linking here.

Reply:
Otherwise, we should probably make a note that `lerna init` should be run on the `packages`(?) level before running this

Reply:
I added a new section that talks about how the walkthrough uses Lerna, but any monorepo solution would work. I now cover how to install Lerna globally on your machine, so that commands will work. I also outline what alternative behavior you would call if using a different monorepo solution.

Using a monorepo vs. package linking is valuable here, as we use some of the project scaffolding that lerna has to create a Vue base project.